src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFModel.java

```diff
@@ -0,0 +1,239 @@
+    private final List<Double> biasRateSamples = new ArrayList<>();
+    private final List<Double> outlierProbabilitySamples = new ArrayList<>();
+    private final List<AFState.MinorFractions> minorFractionsSamples = new ArrayList<>();
+    private final int numSegments;
+
+    public AFModel(final AFData data) {
+        numSegments = data.numSegments();
+        final AFState initialState = new AFInitializer(data).getInitializedState();
+
+        final Sampler<Double, AFState, AFData> meanBiasSampler = new AFSamplers.MeanBiasSampler();
+        final Sampler<Double, AFState, AFData> biasRateSampler = new AFSamplers.BiasRateSampler();
+        final Sampler<Double, AFState, AFData> pOutlierSampler = new AFSamplers.POutlierSampler();
+        final Sampler<AFState.MinorFractions, AFState, AFData> minorFractionsSampler =
+                new AFSamplers.MinorFractionsSampler(data.numSegments());
+
+        model = new ParameterizedModel.GibbsBuilder<>(initialState, data, AFState.class)
+                .addParameterSampler(AFState.MEAN_BIAS_NAME, meanBiasSampler, Double.class)
+                .addParameterSampler(AFState.BIAS_RATE_NAME, biasRateSampler, Double.class)
+                .addParameterSampler(AFState.P_OUTLIER_NAME, pOutlierSampler, Double.class)
+                .addParameterSampler(AFState.MINOR_FRACTIONS_NAME, minorFractionsSampler, AFState.MinorFractions.class)
+                .build();
+    }
+
+    /**
+     * Compute the log-likelihood of a alt reads and r ref reads given minor fraction f and gamma hyperparameters
+     * (specifying the distribution on allelic biases) mu (mean) and beta (rate = mean/variance) and given
+     * an alt minor, ref minor, or outlier indicator state
+     *
+     * Note that this is a collapsed log-likelihood in that the latent variable corresponding to the
+     * allelic bias at this site has been marginalized out.
+     *
+     * See docs/ACS-methods.pdf for derivation.
+     *
+     * Finally, note that this is a static method and does not get mu, beta, and minorFraction from an AFState object
+     * We need such functionality because MCMC evaluates the likelihood under proposed parameter changes
+     *
+     * @param parameters allele fraction parameters at this het site
+     * @param count AllelicCount of alt and ref reads
+     * @param indicator the hidden state (alt minor / ref minor / outlier)
+     *
+     * @return if indicator == ALT_MINOR
+     * log { [beta^alpha / Gamma(alpha)][(1-pi)/2] * int_{0 to infty} f^a * (1-f)^r * lambda^(alpha + r - 1) * exp(-beta*lambda)/(f + (1-f)*lambda)^n d lambda }
+     * if indicator == REF_MINOR
+     * same as ALT_MINOR case but with f <--> 1 - f
+     * if indicator == OUTLIER log {pi * a!r!/(n+1)!}
+     * where alpha = mu*beta and n = a + r
+     */
+    public static double hetLogLikelihood(final AFParameters parameters, final AllelicCount count, final AFIndicator indicator) {
```

Comment:
i think a different (non-overloaded) method name could be warranted here, but i'll leave it up to you.